compiler/rustc_mir/src/transform/check_const_item_mutation.rs

```diff
@@ -31,6 +31,28 @@ impl<'a, 'tcx> ConstMutationChecker<'a, 'tcx> {
             None
         }
     }
+
+    fn is_const_item_without_destructor(&self, local: Local) -> Option<DefId> {
+        let def_id = self.is_const_item(local)?;
+        let mut any_dtor = |_tcx, _def_id| Ok(());
+
+        // We avoid linting mutation of a const item if the const's type has a
```

Comment:
Can you open an issue to track removing this (conditional on having a stable attribute to suppress the lint for a particular `const` item), and add a FIXME?